packages/reactivity/src/collectionHandlers.ts

```diff
@@ -173,6 +173,24 @@ function createForEach(isReadonly: boolean, isShallow: boolean) {
   }
 }
 
+function createSetProtoMethod(method: TriggerOpTypes) {
+  return function (this: SetTypes, value: unknown, _isShallow = false) {
+    if (!_isShallow && !isShallow(value) && !isReadonly(value)) {
+      value = toRaw(value)
+    }
+    const target = toRaw(this)
+    const proto = getProto(target)
+    const hadKey = proto.has.call(target, value)
```

Comment:
I don't understand what this is for. The name `value` here is a bit misleading, as it refers to the other set. Why do we need to check whether one set is inside the other? This seems to break the methods if one set contains the other:

- [Playground - this PR](https://deploy-preview-11399--vue-sfc-playground.netlify.app/#eNp9kjFvwjAQhf+K5YVUomFoJ5QitRVDO7RV6VZ3MMkRDI5t2ZeAFOW/95wowIDYfO99Pj+fruXPzqVNDXzOs5B75ZAFwNothFGVsx5ZyzzIHFUDrGMbbys2IX4ijDC5NQGZZE8nJDFwYCvA5HeyBa3t5O/ubuTW17mD9boYOGFkKosiWffneMtqSLUtE5nWRllDDlnZbEhKGalAqJyWCFQxlhWqWbQtZeq6bBaLS3V9VT33JpvdM2ORUai9MuUIZ7OLV/iUY6BwG1Wmu2ANja6N/QTPbeWUBv/pkBoGweesd6InaRqH915DX8N01PMt5Psr+i4coyb4l4cAvgHBTx5KXwIO9nL1AUc6n8zKFrUm+ob5DTTYOmYcsJfaFBT7guvTvvULQGP4Ccsjggnjp2LQSHY9Lzjtw+uNr5/jPqSP/T1hOt79A7nM1NU=)